ddr_summary_figures.R

```diff
@@ -0,0 +1,220 @@
+# Gregory Way 2017
+# PanCancer Classifier
+# ddr_summary_figures.R
+#
+# Visualize summary for DNA Damage Repair/Response (DDR) classification scores
+#
+# Usage: Run by assigning where the within classifier summary is and where the
+#        Pan Cancer classifier summary is
+#
+#     Rscript --vanilla ddr_summary_figures.R
+#
+# Output:
+# Several figures to summarize DDR findings
+
+library(dplyr)
+library(pheatmap)
+library(ggplot2)
+source("pancancer_util.R")
+
+results <- parse_summary(file.path("classifiers", "TP53",
+                                   "classifier_summary.txt"))
+
+# 1) Heatmap of the distribution of aberrant events across tumors
+heat_df <- readr::read_tsv(file.path("tables",
+                                     "TP53_mutations_heatmap_data.csv"))
+prop_matrix <- as.matrix(heat_df[,c(3, 2)])
+rownames(prop_matrix) <- heat_df$DISEASE
+colnames(prop_matrix) <- c("Loss", "Mutation")
+
+# All diseases that are used in building the classifier
+tp53_dis <- results[["Tissues"]]
+
+# Build a vector for heatmap labels
+classifier <- c()
+for (disease in rownames(prop_matrix)) {
+  if (disease %in% tp53_dis) {
+    classifier <- c(classifier, "Training")
+  } else {
+    classifier <- c(classifier, "Dropped")
+  }
+}
+
+classifier <- data.frame(classifier)
+rownames(classifier) <- rownames(prop_matrix)
+classifier$classifier <- factor(classifier$classifier,
+                                levels = c("Training", "Dropped"))
+
+prop_matrix <- prop_matrix[names(sort(prop_matrix[,2], decreasing = TRUE)), ]
+
+# Plot and save heatmap
+pheatmap(t(prop_matrix * 100), scale = "none", cluster_rows = FALSE,
+         cluster_cols = FALSE,
+         display_numbers = TRUE, number_format = "%.0f", fontsize_number = 8,
+         number_color = "black", annotation_col = classifier,
+         annotation_names_col = F, legend = FALSE,
+         filename = file.path("classifiers", "TP53", "figures",
+                              "tp53_summary_heatmap.pdf"),
+         width = 8, height = 2)
+
+# 2) Coefficients contributing to the model
+coef_df <- results[["Coefficients"]]
+coef_df <- coef_df[, -1]
+coef_df <- coef_df[order(coef_df$weight, decreasing = FALSE), ]
+coef_df$rank <- 1:nrow(coef_df)
+
+p <- ggplot(coef_df, aes(x = 1:nrow(coef_df), y = weight)) +
+  geom_point(aes(fill = "black"), size = 0.01) +
+  base_theme + theme(axis.line.x = element_line(),
+                     axis.line.y = element_line(),
+                     axis.ticks = element_line(),
+                     plot.margin = unit(c(0.25, 0.25, 0.1, 0.1),"cm")) +
+  labs(list(x = "Rank", y = "Weight")) +
+  scale_y_continuous(breaks = seq(-0.25, 0.25, 0.05)) +
+  scale_x_continuous(breaks = seq(0, 8000, 1000)) +
+  geom_segment(aes(x = 0, y = 0, yend = 0, xend = nrow(coef_df)),
+               colour = "red", linetype = "dashed", size = 0.2)
+
+p <- add_arrow_label(p = p, x = 650, y = -0.205, label = "AEN",
```

Comment:
In line78-116, you hard-coded the (x,y) axis for many labels. How do you get these numbers? Could you get these numbers from other places instead of hard-coding them?